pkg/cluster/member.go

```diff
@@ -31,10 +33,16 @@ func (c *Cluster) updateMembers(known etcdutil.MemberSet) error {
 		var name string
 		if c.cluster.Spec.SelfHosted != nil {
 			name = m.Name
-			if len(name) == 0 {
+			if len(name) == 0 || len(m.ClientURLs) == 0 {
```

Comment:
we are going to access m.ClientURL[0] later. we should check if the server finished publishing its client url. or we might panic.